### PR TITLE
bugfix: attach hanging of cri container

### PR DESCRIPTION
**Pull request by YaoZengzeng**
Signed-off-by: YaoZengzeng <yaozengzeng@zju.edu.cn>

### Ⅰ. Describe what this PR did

If we want to get the log of cri container, the step as follows:

1. create container

2. attach to the io of the container

3. start the container

Then the IO of the container will be created in step 2

So we should decide whether open the Stdin of Container IO, otherwise it will never be opened

which will result the hanging of attach

### CODE REVIEW

File: cri/v1alpha1/cri_utils.go [hunk 1]
```diff
@@ -866,6 +866,25 @@ func parseUserFromImageUser(id string) string {
...
+	f, err := os.OpenFile(logPath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0640)
+	if err != nil {
+		return fmt.Errorf("failed to create container for opening log file failed: %v", err)
+	}
+	// Attach to the container to get log.
+	attachConfig := &mgr.AttachConfig{
+		Stdin:      openStdin,
+		Stdout:     true,
+		Stderr:     true,
+		CriLogFile: f,
```
[the remark this answers is not in the document]
Comment (YaoZengzeng) [reply cri/v1alpha1/cri_utils.go]: It will be closed when IO is finished.

[the remark this answers is not in the document]
Comment (fuweid) [reply cri/v1alpha1/cri_utils.go]: could we disable json-file here? I mean that we can disable json-file in other PR.

File: cri/v1alpha1/cri_utils.go [hunk 2]
```diff
@@ -866,6 +866,25 @@ func parseUserFromImageUser(id string) string {
...
+	}
+	err = c.ContainerMgr.Attach(context.Background(), containerID, attachConfig)
+	if err != nil {
+		return fmt.Errorf("failed to attach to container %q to get its log: %v", containerID, err)
```
Comment (fuweid) [inline cri/v1alpha1/cri_utils.go]: how about `pkgerrors.Wrapf`?

Comment (fuweid) [reply cri/v1alpha1/cri_utils.go]: cool

**Pull request merged**